airflow/contrib/hooks/dingding_webhook_hook.py

```diff
@@ -0,0 +1,132 @@
+    :param webhook_token: Dingding webhook token
+    :type webhook_token: str
+    :param message_type: Message type you want to send to Dingding
+                         support text/link/markdown/actionCard/feedCard five types so far
+    :type message_type: str
+    :param message: The message you want to send on Dingding
+    :type message: str or dict
+    :param at_mobiles: Users you want to remind at this message.
+    :type at_mobiles: list of str
+    :param at_all: Should remind all people in group or not. If True, will overwrite ``at_mobiles``
+    :type at_all: bool
+    """
+
+    def __init__(self,
+                 http_conn_id='dingding_webhook_default',
+                 webhook_token=None,
+                 message_type='text',
+                 message=None,
+                 at_mobiles=False,
+                 at_all=False,
+                 *args,
+                 **kwargs
+                 ):
+        super(DingdingWebhookHook, self).__init__(*args, **kwargs)
+        self.http_conn_id = http_conn_id
+        self.endpoint = self._get_endpoint(webhook_token, http_conn_id)
+        self.message_type = message_type
+        self.message = message
+        self.at_mobiles = at_mobiles
+        self.at_all = at_all
+
+    def _get_endpoint(self, manual_token, http_conn_id):
+        """
+        Get Dingding endpoint for sending message. If both ``manual_token`` and
+        ``http_conn_id`` specific, use ``manual_token``
+
+        :param manual_token: Manually provided token
+        :type manual_token: str
+        :param http_conn_id: http_conn_id provided
+        :type http_conn_id: str
+        """
+        if manual_token:
+            token = manual_token
+        elif http_conn_id:
+            conn = self.get_connection(http_conn_id)
+            extra = conn.extra_dejson
+            token = extra.get('webhook_token')
+        else:
+            raise AirflowException('Cannot get token: No valid Dingding '
+                                   'webhook token nor http_conn_id supplied')
+        return 'robot/send?access_token={}'.format(token)
+
+    def _build_message(self):
+        """
+        Build different type of Dingding message
+        As most commonly used type, text message just need post message content
+        rather than a dict like ``{'content': 'message'}``
+        """
+        if self.message_type in ['text', 'markdown']:
+            data = {
+                'msgtype': self.message_type,
+                self.message_type: {
+                    'content': self.message
+                } if self.message_type == 'text' else self.message,
+                'at': {
+                    'atMobiles': self.at_mobiles,
+                    'isAtAll': self.at_all
+                }
+            }
+        else:
+            data = {
+                'msgtype': self.message_type,
+                self.message_type: self.message
+            }
+        return json.dumps(data)
+
+    def send(self):
```

Comment:
```suggestion
    def send_message(self):
```

Reply:
I think is a monor change, I will keep the old function name.